### DISCUSSION TITLE
Deploy on Release

### DISCUSSION THREAD

**Discussion by Gmanicus** [coollabsio/coolify#3471] [Improvement Requests]
Hi there,

It would be helpful to be able to deploy my application only _after_ I create a new release on my Github repo. That way I make sure I properly document every release and then it becomes a requirement for the changes to hit production. Currently, the only options seem to be

- Release still optional
  (1) deploy automatically on push to main
  (2) manually deploy
- Release required
  (3) create a custom webhook to do this.

**Comment by mt-bt** [comment]
Strong +1

**Comment by ChristopheCVB** [comment]
I would say, it would be cool to be able to deploy on a tag regex, example `release/*`

This is a duplicate of https://github.com/coollabsio/coolify/discussions/4532

**Comment by robinsimonklein** [comment]
This could really be a great behavior!

**Comment by francosang** [comment]
I am doing workarounds with different branches to achieve this missing feature.

**Reply by ChristopheCVB** [reply]
I've accomplished it with GitHub actions and workflows

**Reply by francosang** [reply]
I was using the main branch to deploy automatically to my staging/dev environment, and then I also have a releases branch in which I merge main when it gets stable enough that triggers a deploy to the prod env.

How is your approach?

**Reply by ChristopheCVB** [reply]
I have a `main` branch that gets auto deployed on push (normal Coolify integration) on a `develop` env and then i have [Release Please](https://github.com/googleapis/release-please) plugged into that branch that creates a release when I merge the Release Please PR created the the [Release Please Github Action](https://github.com/googleapis/release-please-action), and then, a workflow that triggers when a release is created that deploys with a [Github Action](https://github.com/marketplace/actions/coolify-application-deploy) on a Coolify env, `production`, that has auto deploy disabled

**Reply by ChristopheCVB** [reply]
Here is a [Gist](https://gist.github.com/ChristopheCVB/63f05301328dc3a31e0d6ab8cfb65a55) of it

**Comment by amiceli** [comment]
Any news ? It will be useful to deploy when a release is created on Github.

**Reply by ChristopheCVB** [reply]
Use this Gist:
https://gist.github.com/ChristopheCVB/63f05301328dc3a31e0d6ab8cfb65a55#file-github_workflows_deploy-yml
